pkg/builder/controller.go

```diff
@@ -58,15 +60,25 @@ type Builder struct {
 	ownsInput        []OwnsInput
 	watchesInput     []WatchesInput
 	mgr              manager.Manager
+	cluster          cluster.Cluster
+	logicalName      logical.Name
 	globalPredicates []predicate.Predicate
 	ctrl             controller.Controller
 	ctrlOptions      controller.Options
 	name             string
 }
 
+func (blder *Builder) clone() *Builder {
+	clone := *blder
+	clone.cluster = nil
+	clone.logicalName = ""
+	clone.ctrl = nil
+	return &clone
+}
+
 // ControllerManagedBy returns a new controller builder that will be started by the provided Manager.
 func ControllerManagedBy(m manager.Manager) *Builder {
-	return &Builder{mgr: m}
+	return &Builder{cluster: m, mgr: m}
```

Comment:
I agree, this might be a deeper change. Should we opt to expose the Cluster on the manager via different means?

For example, let's say we deprecate the mgr.GetClient/GetCache/GetScheme, etc on the Manager; would we have something like `GetDefaultCluster() cluster.Cluster` which then exposes the same functions? Happy to wordsmith the naming of everything, but it'd be good to know if we should actually propose this change or not.

cc @alvaroaleman